crates/runtime/tests/models/hf.rs

```diff
@@ -222,6 +234,106 @@ async fn huggingface_test_embeddings() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[tokio::test]
+async fn huggingface_test_chat_completion() -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(None);
+
+    let mut model_with_tools =
+        get_huggingface_model("microsoft/Phi-3-mini-4k-instruct", "phi3", "hf_model");
+    model_with_tools
+        .params
+        .insert("spice_tools".to_string(), "auto".into());
+
+    let app = AppBuilder::new("text-to-sql")
+        .with_dataset(get_taxi_trips_dataset())
+        .with_model(model_with_tools)
+        .build();
+
+    let api_config = create_api_bindings_config();
+    let http_base_url = format!("http://{}", api_config.http_bind_address);
+    let rt = Arc::new(Runtime::builder().with_app(app).build().await);
+
+    let rt_ref_copy = Arc::clone(&rt);
+    tokio::spawn(async move {
+        Box::pin(rt_ref_copy.start_servers(api_config, None, EndpointAuth::no_auth())).await
+    });
+
+    tokio::select! {
+        // increased timeout to download and load huggingface model
+        () = tokio::time::sleep(std::time::Duration::from_secs(300)) => {
+            return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+        }
+        () = rt.load_components() => {}
+    }
+
+    let _response = send_chat_completions_request(
+        http_base_url.as_str(),
+        vec![
+            ("system".to_string(), "You are an assistant that responds to queries by providing only the requested data values without extra explanation.".to_string()),
+            ("user".to_string(), "Provide the total number of records in the taxi_trips dataset. If known, return a single numeric value.".to_string()),
+        ],
+        "hf_model",
+        false,
+    ).await?;
+
+    // Result verification is disabled due to issue below: model does not use tools and can't provide the expected response.
+    // https://github.com/spiceai/spiceai/issues/3426
+    // assert_eq!(response["choices"][0]["message"]["content"].as_str(), Some("10"));
```

Comment:
Should we assert on a snapshot with the content replaced, like the below test?

Reply:
That makes sense - thank you, added snapshost based verification step for both HF and OpenAI:
https://github.com/spiceai/spiceai/pull/3433/commits/73e30f953e322b14c293f6c8f20d1d31c12491c9